### ISSUE TITLE
Inconsistency in user creation process

### ISSUE THREAD

**Issue by dsever** [DefectDojo/django-DefectDojo#4547]
There is no equal implementation for users created by API and one created by GUI.

Clarification:
During the creation of user in GUI,  program set `unusable`  (set_unusable_password() ) password that brings advantage when external identity provider is in use (e.g. LDAP), so user can't change their password and use it as local user. 

```
    if request.method == 'POST':
        form = AddDojoUserForm(request.POST)
        contact_form = UserContactInfoForm(request.POST)
        if form.is_valid() and contact_form.is_valid():
            user = form.save(commit=False)
            user.set_unusable_password()
            user.active = True
            user.save()
```

At the other hand API doesn't respect this best practice, so unusable password is not there, so password is blank, but after user will login using external provider, will be able to change his password and use it as local user.

References:
https://docs.djangoproject.com/en/3.2/ref/contrib/auth/#django.contrib.auth.models.User.set_unusable_password
https://stackoverflow.com/questions/50374865/django-1-11-authentication-set-unusable-password-vs-setting-password-to-none

**Comment by StefanFl** [comment]
I can fix this together with #4532

**Comment by valentijnscholten** [comment]
Thanks for reporting/fixing. Actually didn't notice Defect Dojo doesn't have a "Forgot Password" feature. Should we add it (behind a settings flag)?

**Comment by valentijnscholten** [comment]
Users also get created via LDAP (and possibly OIDC/Oauth/SAML), not sure if the unusable password is already being set there.

**Comment by StefanFl** [comment]
I don't see how LDAP and OAuth create users, maybe it's part of the libraries. With OAuth for Microsoft Azure in my productive environment I can't change my password, so I assume it's fine there.

**Comment by valentijnscholten** [comment]
SAML: https://github.com/fangli/django-saml2-auth/blob/825c1092803c38f626089763a7a3e030f2f82c1d/django_saml2_auth/views.py#L140
LDAP: https://github.com/django-auth-ldap/django-auth-ldap/blob/2758cdb1700933aacb0c0a79847ec9923d095cd3/django_auth_ldap/backend.py#L596
Social App Django: https://github.com/python-social-auth/social-app-django/blob/b6e791adf96c3f13c3e499669e50093f8dc23461/social_django/storage.py#L59

Looks like only LDAP sets an unusable passwords for newly created users.

**Comment by StefanFl** [comment]
For SAML we would need someone else to test it, I don't have the environment available. 
Don't know why, but for Social App Django it works in my productive environment. Users who login via Microsoft can't change their password.

**Comment by dsever** [comment]
@StefanFl  for SAML implementation is ok, there is unusable password in place, and user can't do it.

**Comment by valentijnscholten** [comment]
fixed in dev / 2.0.0
